README.md

```diff
@@ -1,116 +1,165 @@
 
-#### 1. **Setting Up a Basic Model**:
+## Installation and Setup
 
-This snippet shows how to initialize a neural network with basic hyperparameters.
+To get started with this project, first clone this repository using the following command:
 
-```python
-import torch
-import torch.nn as nn
-from your_project_model import RetNetModel  # Replace with your actual model import
+```bash
+git clone https://github.com/DRAGNLabs/301r_retnet.git
+cd 301r_retnet
+```
+
+### Python Version Requirement
+
+Ensure you have Python 3.11 installed. If you do not have Python 3.11, you can download it from the official Python website or use a package manager.
+
+### Preparing the Environment
+Optionally, create an environement. Then install all the necessary dependencies:
+```bash
+# Optionally create a new Mamba environment with Python 3.11 and specify a name
+mamba create -n <YOUR_ENV_NAME> python=3.11
 
-# Basic hyperparameters
-hyperparams = {
-    "learning_rate": 0.001,
-    "batch_size": 64,
-    "num_layers": 4,
-    "embed_dim": 256,
-    "num_heads": 8,
-    "dropout_rate": 0.1
-}
+# Activate the Mamba environment
+mamba activate <YOUR_ENV_NAME>
 
-model = RetNetModel(num_layers=hyperparams["num_layers"],
-                    embed_dim=hyperparams["embed_dim"],
-                    num_heads=hyperparams["num_heads"],
-                    dropout_rate=hyperparams["dropout_rate"])
-optimizer = torch.optim.Adam(model.parameters(), lr=hyperparams["learning_rate"])
+pip install -r requirements.txt
 ```
 
-#### 2. **Implementing Learning Rate Scheduler**:
+### Preparing YAML Configuration Files
 
-An example of using a learning rate scheduler.
+This project uses YAML configuration files to store all pipeline parameters and paths. The design choice of the YAML file is intended to eliminate repetition of commonly used parameters across code, as well as simplify future changes and refactors, allow developers to add new parameters, and make all settings visible to the user in one consolidated place.
 
-```python
-scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=10, gamma=0.1)
+To prepare a YAML config file, copy [template_config.yaml](./configs/template_config.yaml) into the [user_configs](./configs/user_configs/) folder. Fill out all parameters accordingly. Absolute paths are preferred for any path variables, but the repository is setup to work flexibly with any desired directory structure.
 
-# Training loop
-for epoch in range(num_epochs):
-    # Training process
-    ...
-    scheduler.step()  # Update learning rate
-```
+*Note:* In most cases, YAML does not expect strings. Adding quotation marks around argurments in the config file can lead to unexpected errors.
```

Comment:
Is this verified or only a one-off issue we saw on @jtappen1's setup?

Reply:
@JayOrten, I don't think you need quotation marks in YAML, right?

Reply:
No, it ruined mine as well. Jay said we yaml doesn't expect quotation marks.